pallets/settlement-risc0/src/lib.rs

```diff
@@ -84,11 +90,11 @@ pub mod pallet {
     ) -> Result<(), Error<T>> {
         log::trace!("Checking size");
         ensure!(
-            (proof).len() <= MAX_PROOF_SIZE,
+            (proof).len() <= T::MaxProofSize::get().try_into().unwrap(),
             Error::<T>::InvalidProofSize
         );
         ensure!(
-            (pubs).len() <= MAX_PUBS_SIZE,
+            (pubs).len() <= T::MaxPubsSize::get().try_into().unwrap(),
             Error::<T>::InvalidPublicInputsSize
         );
```

Comment:
```suggestion
        ensure!(
            (proof).len() <= T::MaxProofSize::get() as usize,
            Error::<T>::InvalidProofSize
        );
        ensure!(
            (pubs).len() <= T::MaxPubsSize::get() as usize,
            Error::<T>::InvalidPublicInputsSize
        );
```
From `u32` to `usize` you can use `as` directly

Reply:
Modified as suggested.